### PR TITLE
Add support for Debian 12

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

**Comment by evgeni** [comment]
Why did CI not run on D12?

File: manifests/params.pp [hunk 1]
```diff
@@ -11,7 +11,7 @@
   $repo_proxy_password         = 'absent'
   $repo_proxy_username         = 'absent'
   $repo_release                = undef
-  $repo_url_suffix             = '12.x'
+  $repo_url_suffix             = '18.x'
```
Comment (bastelfreak) [inline manifests/params.pp:14]: Is that a breaking change?

Comment (evgeni) [reply manifests/params.pp:14]: (I'll probably split it out to an own PR, once I verified it fixes what I think it should fix on D12/U22)

File: manifests/params.pp [hunk 2]
```diff
@@ -27,7 +27,10 @@
       $nodejs_dev_package_ensure = 'absent'
       $nodejs_package_name       = 'nodejs'
       $npm_package_ensure        = 'absent'
-      $npm_package_name          = 'npm'
+      $npm_package_name          = $nodejs::manage_package_repo ? {
+        true    => false,
+        default => 'npm'
+      }
```
Comment (evgeni) [inline manifests/params.pp:33]: this produces an interesting circular thingy, which seems to accidentally work on p7 but not on p8

however, using `$manage_package_repo` here is useless, as that is not updated when the users sets `nodejs::manage_package_repo`.

HALP!?

**Review by bastelfreak — APPROVED** [review]

**Pull request merged**